Upgrade dependencies

notlee opened the pull request:
**ready for a preliminary review but the build will fail until [colour usecases may be set as hex values](https://github.com/Financial-Times/o-colors/pull/231) and [custom links work with mixes](https://github.com/Financial-Times/o-typography/pull/234)**

origamiserviceuser commented:
o-teaser bundle size difference from 3.5.9 to 5bdef9dfce8e077667651f0662fcf4ca1f70a40a
css: 1.41kb increase (0.13kb/gzip)

notlee commented:
> I've fixed a bunch of things based on the demos being different to 3.x demos, I'd quite like your eyes on them @notlee if you get a chance. All of the dependencies are updated, and the only thing that needs addressing is a contrast issue that's failing the Pa11y tests.
> 
> The colour in `$_o-teaser-muted` is not accessible on paper or wheat. After digging around, I discovered it's different from the colour used in the 3.x codebase, the output has changed:
> 
> ```diff
> - #66605c
> + #746e68
> ```
> 
> I think this is because we're setting the `o-teaser/base` `text` colour to the result of `oColorsGetTextColor('wheat')`. So the base colour is accessible on wheat just fine. However then `$_o-teaser-muted` takes this colour and lightens it by 60%, making it not accessible. I'm not sure what the best way to fix this is, I'm tired and can't look at it any more 😅
> 
> The way I tested all of the teaser demos may be helpful to you. I threw together and HTML file that displays the local 4.x demos side-by-side with the Build Service 3.x demos. It made the process of checking far quicker:
> **Useful HTML file for comparing teaser demos**

very useful! I think I've fixed it based on [oColorsFor previous behavior](https://github.com/Financial-Times/o-colors/blob/v4.10.3/src/scss/_mixins.scss#L80)
